--- rdf-canonize4.0.1/lib/index.js
+++ rdf-canonize5.0.0/lib/index.js
@@ -37,12 +37,6 @@
 const RDFC10 = require('./RDFC10');
 const RDFC10Sync = require('./RDFC10Sync');
 
-// optional native support
-let rdfCanonizeNative;
-try {
-  rdfCanonizeNative = require('rdf-canonize-native');
-} catch(e) {}
-
 // return a dataset from input dataset or n-quads
 function _inputToDataset(input, options) {
   if(options.inputFormat) {
@@ -79,20 +73,6 @@
 // expose helpers
 exports.NQuads = require('./NQuads');
 exports.IdentifierIssuer = require('./IdentifierIssuer');
-
-/**
- * Get or set native API.
- *
- * @param {object} [api] - The native API.
- *
- * @returns {object} - The currently set native API.
- */
-exports._rdfCanonizeNative = function(api) {
-  if(api) {
-    rdfCanonizeNative = api;
-  }
-  return rdfCanonizeNative;
-};
 
 /**
  * Asynchronously canonizes an RDF dataset.
@@ -118,7 +98,6 @@
  *     falsy for a JSON dataset.
  *   {string} [format] - The format of the output. Omit or use
  *     'application/n-quads' for a N-Quads string.
- *   {boolean} [useNative=false] - Use native implementation.
  *   {number} [maxWorkFactor=1] - Control of the maximum number of times to run
  *     deep comparison algorithms (such as the N-Degree Hash Quads algorithm
  *     used in RDFC-1.0) before bailing out and throwing an error; this is a
@@ -147,19 +126,6 @@
 exports.canonize = async function(input, options = {}) {
   const dataset = _inputToDataset(input, options);
   _checkOutputFormat(options);
-
-  if(options.useNative) {
-    if(!rdfCanonizeNative) {
-      throw new Error('rdf-canonize-native not available');
-    }
-    if(options.createMessageDigest) {
-      throw new Error(
-        '"createMessageDigest" cannot be used with "useNative".');
-    }
-    return new Promise((resolve, reject) =>
-      rdfCanonizeNative.canonize(dataset, options, (err, canonical) =>
-        err ? reject(err) : resolve(canonical)));
-  }
 
   if(!('algorithm' in options)) {
     throw new Error('No RDF Dataset Canonicalization algorithm specified.');
@@ -202,7 +168,6 @@
  *     falsy for a JSON dataset.
  *   {string} [format] - The format of the output. Omit or use
  *     'application/n-quads' for a N-Quads string.
- *   {boolean} [useNative=false] - Use native implementation.
  *   {number} [maxWorkFactor=1] - Control of the maximum number of times to run
  *     deep comparison algorithms (such as the N-Degree Hash Quads algorithm
  *     used in RDFC-1.0) before bailing out and throwing an error; this is a
@@ -236,16 +201,6 @@
   const dataset = _inputToDataset(input, options);
   _checkOutputFormat(options);
 
-  if(options.useNative) {
-    if(!rdfCanonizeNative) {
-      throw new Error('rdf-canonize-native not available');
-    }
-    if(options.createMessageDigest) {
-      throw new Error(
-        '"createMessageDigest" cannot be used with "useNative".');
-    }
-    return rdfCanonizeNative.canonizeSync(dataset, options);
-  }
   if(!('algorithm' in options)) {
     throw new Error('No RDF Dataset Canonicalization algorithm specified.');
   }
